lib/instrumentation/trace-context.js

```diff
@@ -0,0 +1,109 @@
+'use strict'
+
+const crypto = require('crypto')
+
+const SIZES = {
+  version: 1,
+  traceId: 16,
+  id: 8,
+  flags: 1,
+  parentId: 8,
+
+  // Aggregate sizes
+  ids: 24, // traceId + id
+  all: 34
+}
+
+const OFFSETS = {
+  version: 0,
+  traceId: SIZES.version,
+  id: SIZES.version + SIZES.traceId,
+  flags: SIZES.version + SIZES.ids,
+
+  // Additional parentId is stored after the header content
+  parentId: SIZES.version + SIZES.ids + SIZES.flags
+}
+
+function defineLazyProp (obj, prop, fn) {
+  Object.defineProperty(obj, prop, {
+    configurable: true,
+    enumerable: true,
+    get () {
+      const value = fn()
+      Object.defineProperty(obj, prop, {
+        enumerable: true,
+        value
+      })
+      return value
+    }
+  })
+
+  return obj
+}
+
+function hexSliceFn (buffer, offset, length) {
+  return () => buffer.slice(offset, length).toString('hex')
+}
+
+function maybeHexSliceFn (buffer, offset, length) {
+  const fn = hexSliceFn(buffer, offset, length)
+  return () => {
+    const value = fn()
+    if (/[1-9a-f]/.test(value)) {
```

Comment:
Consider adding a comment saying that this is to check that the value isn't all zeros. I got confused and thought that you wanted to check that it was valid hex

Reply:
Will do. 👍